src/openMVG/sfm/pipelines/hierarchical_hyper/scene_aligner.cpp

```diff
@@ -24,28 +25,113 @@ namespace sfm {
 using namespace openMVG::cameras;
 using namespace openMVG::geometry;
 
+std::shared_ptr<cameras::IntrinsicBase> findBestIntrinsic(const SfM_Data& sfm_data_A, const SfM_Data& sfm_data_B, openMVG::IndexT cam_id)
+{
+  // both alternatives for the intrinsic
+  const auto intrinsicA = sfm_data_A.intrinsics.at(cam_id);
+  const auto intrinsicB = sfm_data_B.intrinsics.at(cam_id);
+
+  // quick escape in case intrinsics are equal
+  // TODO ? here we only check for pointer equality should we do more ?
```

Comment:
Each intrinsic have a HASH build on the parameter.
So you can check that the camera type and the hash is the same:
`getType && hashValue`

Reply:
ah yes that's what I wanted ! do I need to check for both though ? it looks like the type information is contained in the hash ?